lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+    },
+    tlsca: {
+      format: '*',
+      env: 'CMD_LDAP_TLS_CA',
+      default: undefined
+    },
+    tlsOptions: {
+      format: '*',
+      default: undefined
+    }
+  },
+  saml: {
+    idpSsoUrl: {
+      doc: 'Login URL for the SAML Identity Provider',
+      format: '*',
+      env: 'CMD_SAML_IDPSSOURL',
+      default: undefined
+    },
+    idpCert: {
+      doc: 'SAML Identity Provider certificate',
+      format: '*',
+      env: 'CMD_SAML_IDPCERT',
+      default: undefined
+    },
+    issuer: {
+      doc: 'SAML Identity Provider Identifier issuer',
+      format: '*',
+      env: 'CMD_SAML_ISSUER',
+      default: undefined
+    },
+    identifierFormat: {
+      doc: 'SAML Identity Provider format',
+      env: 'CMD_SAML_IDENTIFIERFORMAT',
+      default: 'urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress'
+    },
+    disableRequestedAuthnContext: {
+      doc: 'SAML if truthy, do not request a specific authentication context. This is known to help when authenticating against Active Directory (AD FS) servers.',
+      format: 'Boolean',
+      env: 'CMD_SAML_DISABLEREQUESTEDAUTHNCONTEXT',
+      default: false
+    },
+    groupAttribute: {
+      format: '*',
+      env: 'CMD_SAML_GROUPATTRIBUTE',
+      default: undefined
+    },
+    externalGroups: {default: []},
+    requiredGroups: {default: []},
+    attribute: {
+      id: {
+        format: '*',
+        env: 'CMD_SAML_ATTRIBUTE_ID',
+        default: undefined
+      },
+      username: {
+        format: '*',
+        env: 'CMD_SAML_ATTRIBUTE_USERNAME',
+        default: undefined
+      },
+      email: {
+        format: '*',
+        env: 'CMD_SAML_ATTRIBUTE_EMAIL',
+        default: undefined
+      }
+    }
+  },
+  email: {
+    doc: 'Allow users to sign-in using their E-Mail address.',
+    default: true,
+    env: 'CMD_EMAIL',
+    format: 'Boolean'
+  },
+  allowEmailRegister: {
+    doc: 'Allow users to sign up using their E-Mail. Requires email to be set to true.',
+    default: true,
+    env: 'CMD_ALLOW_EMAIL_REGISTER',
+    format: 'Boolean'
+  },
+  allowGravatar: {
+    doc: 'Allow the usage of Libravatar for user profile pictures.',
+    default: true,
+    env: 'CMD_ALLOW_GRAVATAR',
+    format: 'Boolean'
+  },
+  allowPDFExport: {
+    doc: 'Allow PDF export documents as PDF.',
+    default: true,
+    env: 'CMD_ALLOW_PDF_EXPORT',
+    format: 'Boolean'
+  },
+  openID: {
+    doc: 'Allow OpenID as authentication method.',
+    default: false,
+    env: 'CMD_OPENID',
+    format: 'Boolean'
+  },
+  linkifyHeaderStyle: {
+    doc: 'Default linking style for headers used by the markdown rendering engine for notes',
+    format: function (val) {
+      if (!/(keep-case|gfm)/.test(val)) {
+        throw new Error('Value has to be \'keep-case\' or \'gfm\'')
+      }
+    },
+    default: 'keep-case'
+  },
+  version: {
+    doc: 'Version number of CodiMD',
+    format: 'semver',
+    default: null
+  },
+  minimumCompatibleVersion: {
```

Comment:
this file is getting exceedingly long. Maybe we should break it into a few sections? I tried something similar, when I rewrote the two readme files, look at the h2 headings for what I mean by beaking it up: https://github.com/codimd/server/blob/master/docs/configuration-env-vars.md

Reply:
Yes, we should do this. For now, this was the first working version I wanted to provide in order to make it possible to use those.

In the next iteration, I would like to try to separate config schemas for modules, from the core ones. This should already shorten the file a lot.